src/app/posts.services.ts

```diff
@@ -25,7 +25,9 @@ export class PostsService implements CanActivate {
     // auth
     auth_token = '4241b401-7236-493e-92b5-b72696b9d853';
     session_id = null;
-    httpOptions = null;
+    httpOptions: {
```

Comment:
Why this change?

Reply:
Two parts:
- This field is set in the constructor, so the default isn't needed.
- Angular uses the type of `httpOptions.responseType` to decide on the `get`/`post` return type. When this is set to `any`, `get`/`post` defaults to ArrayBuffer instead of JSON.